test/e2e/operator_test.go

```diff
@@ -101,6 +122,72 @@ func TestDefaultClusterIngressExists(t *testing.T) {
 	}
 }
 
+func TestClusterIngressControllerCreateDelete(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ci := &ingressv1alpha1.ClusterIngress{}
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: "default"}, ci); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get default IngressController: %v", err)
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Fatalf("failed to get DNS 'cluster': %v", err)
+	}
+
+	ing := newIngressController(ingressControllerName, ns, ingressControllerName+"."+dnsConfig.Spec.BaseDomain)
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Create(context.TODO(), ing); err != nil {
+			if errors.IsAlreadyExists(err) {
+				return true, nil
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("failed to create IngressController %s/%s: %v", ing.Namespace, ing.Name, err)
+	}
+
+	// Verify the ingress controller deployment created the specified
+	// number of pod replicas.
+	err = wait.PollImmediate(1*time.Second, 60*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ing.Namespace, Name: ing.Name}, ing); err != nil {
+			if ing.Status.Replicas != ing.Spec.Replicas {
+				return false, nil
+			}
+			return true, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("failed to reconcile IngressController %s/%s: %v", ing.Namespace, ing.Name, err)
+	}
+
+	// Verify the ingress controller can be deleted.
+	err = wait.PollImmediate(1*time.Second, 15*time.Second, func() (bool, error) {
+		if err := cl.Delete(context.TODO(), ing); err != nil {
```

Comment:
Similar to above comment, I don't think polling is needed here

Reply:
We need polling to make sure it's gone, but it would make sense to `Delete` outside the `PollImmediate` and do a `Get` inside the `PollImmediate`.

Reply:
@pravisankar I thought the poll was needed b/c the delete is not immediate due to the finalizer.
@Miciah We perform a Get at L165 and compare spec<>status replicas. Why is a 2nd get needed?

Reply:
I mean that as part of the test, we should make sure that the `Delete` call actually eventually deletes the resource.  Does that seem reasonable? If so, we can either repeat the `Delete` call till the resource is gone (in which case we should be checking here for a not-found error value), or we can call `Delete` once and then repeat a `Get` call till the resource is gone.

Reply:
@Miciah thanks for the additional explanation. I now understand your perspective. I'll update the `Delete`.